src/Functions/FunctionsOpDate.cpp

```diff
@@ -0,0 +1,104 @@
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+    size_t getNumberOfArguments() const override { return 2; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (!isDateOrDate32(arguments[0].type) && !isDateTime(arguments[0].type) && !isDateTime64(arguments[0].type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal type {} of 1st argument of function {}. Should be a date or a date with time",
+                arguments[0].type->getName(),
+                getName());
+
+        if (!isInterval(arguments[1].type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal type {} of 2nd argument of function {}. Should be an interval",
+                arguments[1].type->getName(),
+                getName());
+
+        auto op = FunctionFactory::instance().get(Op::internal_name, context);
+        auto op_build = op->build(arguments);
+
+        return op_build->getResultType();
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0, 2}; }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        if (!isDateOrDate32(arguments[0].type) && !isDateTime(arguments[0].type) && !isDateTime64(arguments[0].type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal type {} of 1st argument of function {}. Should be a date or a date with time",
+                arguments[0].type->getName(),
+                getName());
+
+        if (!isInterval(arguments[1].type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal type {} of 2nd argument of function {}. Should be an interval",
+                arguments[1].type->getName(),
+                getName());
+
+        auto op = FunctionFactory::instance().get(Op::internal_name, context);
+        auto op_build = op->build(arguments);
+
+        auto res_type = op_build->getResultType();
+        return op_build->execute(arguments, res_type, input_rows_count);
+    }
+
+private:
+    ContextPtr context;
+};
+
+}
+
+struct AddDate
+{
+    static constexpr auto name = "addDate";
+    static constexpr auto internal_name = "plus";
+};
+
+struct SubDate
+{
+    static constexpr auto name = "subDate";
+    static constexpr auto internal_name = "minus";
+};
+
+using FunctionAddDate = FunctionOpDate<AddDate>;
+using FunctionSubDate = FunctionOpDate<SubDate>;
+
+REGISTER_FUNCTION(AddInterval)
+{
+    factory.registerFunction<FunctionAddDate>();
+    factory.registerFunction<FunctionSubDate>();
```

Comment:
For compat with MySQL's stupid 80s-style upper-case: could you either add the case-insensitive flag to l. 100/101 or add aliases for `ADDDATE()` and `SUBDATE()`?

(please also document these and perhaps even add a test which checks that the aliases work)